controllers/kyma_controller.go

```diff
@@ -148,16 +155,12 @@ func (r *KymaReconciler) getSyncedContext(ctx context.Context, kyma *v1beta2.Kym
 	ctxWithSync, err := remote.InitializeSyncContext(ctx, kyma,
 		r.RemoteSyncNamespace, remoteClient, r.RemoteClientCache)
 	if err != nil {
-		if !kyma.DeletionTimestamp.IsZero() && errors.Is(err, remote.ErrAccessSecretNotFound) {
-			logger.Info("access secret not found for kyma, assuming already deleted cluster")
-			return nil, r.removeAllFinalizersAndUpdateKyma(ctx, kyma)
-		}
 		if util.IsConnectionRefused(err) {
```

Comment:
I think this code goes agains "Single responsibility principle". The "getSyncedContext" function should be focused only on getting synced context, not on handling special cleanup cases, removing cache entries etc.
I think moving the `kyma.DeletionTimestamp.IsZero` away from this function is a good thing, and I'd also move **this** block out of it. Otherwise our logic is scattered across different functions and it's really hard to answer the question:
`What is controller logic in case of a remote connection problem?`

Reply:
make sense, changed accordingly.